### ISSUE TITLE
Boost difference issue

### ISSUE THREAD

**Issue by ggartside** [NREL/OpenStudio#4804]
Issue overview
--------------
[
  [
    [4.572, 0, 0], 
    [0, 0, 0], 
    [0, 0.3048, 0], 
    [4.572, 0.304799, 0], 
  ],
  [
    [-13.4112, 0.3048, 0], 
    [5.1816, 0.3048, 0], 
    [5.1816, 0, 0], 
    [-10.668, -2.66454e-15, 0], 
  ],
]

Subtracting the first polygon from the second results in of 0 polygons

Input polygons:
![image](https://user-images.githubusercontent.com/21084412/221038187-68086dde-fbaf-4c0d-b11d-5a7896022098.png)

## Current Behavior
The subtraction results in 0 polygons

## Expected Behavior
Expected result:
![image](https://user-images.githubusercontent.com/21084412/221038290-9400431e-e17f-442a-99d3-afb1b9e8d28d.png)

## Steps to Reproduce

1. Running regression test test_intersect_136_osm
2. Check the log file for the following error
3. [openstudio.model.Surface] <1> Initial area of other surface 'Surface 109' 5.24902 does not equal post intersection area 1.39354
4.
5.

### Environment

Some additional details about your environment for this issue (if relevant):
 - Platform (Operating system, version):
 - Version of OpenStudio (if using an intermediate build, include SHA):

**Comment by tijcolem** [comment]
Thanks for reporting @ggartside I think we should open an issue with https://github.com/boostorg/geometry is this is bug associates with boost.

**Comment by ggartside** [comment]
@tijcolem I think I may have found a solution for this. I was reading through a similar bug report on boost geometry repository,  https://github.com/boostorg/geometry/issues/1034

In this bug report the developer replied to the initial report as follows

> I can reproduce it, and I am looking into it. If I make all coordinates 10 times larger, the output looks similar but is valid. My theory is that because of the intersection points (which are rounded to the integer grid), in some cases invalid parts can appear. I will look further into this next week, but this is not easy to fix.

My reading of this is that their coordinates are rounded to an integer value, integer grid, and if he had scaled the input values by 10, effectively increasing the resolution of the integer grid, and achieved a better result by doing that and given that all our input values are in meters and so are relatively small numbers, why not try scaling the input values by 1000 to make them millimeters?

So I tried that and scaled the coordinates up in `boostPointFromPoint3d` and back down in `verticesFromBoostPolygon` and then ran my test and the result is that the original issue with subtraction has been resolved and now there are no Initial Area differs errors in the output.

**Comment by ggartside** [comment]
Further development. I found a couple more places where length values need to be scaled and now all the unit tests run clean, however regression test 356 is now failing so I looked in to this further.

We have these two polygons that we are going to intersect
![image](https://user-images.githubusercontent.com/21084412/222219879-8290e7fa-4829-4594-a5b9-6b8482bef10e.png)
They appear to be maybe just touching if the vertex on the green polygon is co-linear with the edge on the blue polygon, will have to figure out if it is on the line, just outside or just inside. However the intersection result from boost is
![image](https://user-images.githubusercontent.com/21084412/222220283-4197fbe0-0a82-471d-b72c-607521dcf1b4.png)
Which is definitely not what we would expect at all

So either way with our without scaling we have a boost boolean operation failure which is unfortunate. I'm going to log both issues with boost now to see if we get any traction from them
